cmd/cluster-network-operator/main.go

```diff
@@ -102,16 +103,29 @@ func main() {
 		log.Fatal(err)
 	}
 
-	// Setup all Controllers
+	stopCh := signals.SetupSignalHandler()
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		<-stopCh
+		cancel()
+	}()
+	defer cancel()
+
+	log.Print("Starting the Cmd.")
+	func() {
+		if err := connectivitycheck.Start(ctx, mgr.GetConfig()); err != nil {
```

Comment:
Why does this have to be started explicitly from `main()` rather than being started like all the other controllers?

If the problem is that it needs access to a `Context` but we don't currently pass that to the Controller start funcs, then just fix the Controller start func code

Reply:
Because it's a library-go operator and CNO is controller-runtime, thus the split. Sure, I could 'try' to make the operator play as a controller-runtime controller but that would get messy.
Having said that, this will probably change now since @squeed change for making CNO a library-go controller is in flight and this PR is probably better off rebased when that merges.